editor/import/editor_import_collada.cpp

```diff
@@ -681,7 +681,8 @@ Error ColladaImport::_create_mesh_surfaces(bool p_optimize, Ref<EditorSceneImpor
 
 				int vertex_index = p.indices[src + vertex_ofs]; //used for index field (later used by controllers)
 				int vertex_pos = (vertex_src->stride ? vertex_src->stride : 3) * vertex_index;
-				ERR_FAIL_INDEX_V(vertex_pos, vertex_src->array.size(), ERR_INVALID_DATA);
+				ERR_FAIL_INDEX_V(vertex_pos + 0, vertex_src->array.size(), ERR_INVALID_DATA);
+				ERR_FAIL_INDEX_V(vertex_pos + 2, vertex_src->array.size(), ERR_INVALID_DATA);
```

Comment:
A simpler check might be:
```suggestion
				ERR_FAIL_INDEX_V(vertex_pos, vertex_src->array.size() - 2, ERR_INVALID_DATA);
```
no?

Reply:
My intention was to always check the upper and lower bound when accessing a sequence of elements. Doing it for every element in the sequence doesn't scale well, and might not catch any additional errors to just checking upper+lower.

With the suggested variant a problems arises if the array size is < 2. By `size()` returning an unsigned value, an underflow can occur when subtracting a value. A out-of-bounds access might therefore not be caught.